packages/camera/android/src/main/java/io/flutter/plugins/camera/CameraPlugin.java

```diff
@@ -47,33 +48,34 @@
 
 public class CameraPlugin implements MethodCallHandler {
 
-  private static final int cameraRequestId = 513469796;
-  private static final SparseIntArray ORIENTATIONS = new SparseIntArray();
-  private static CameraManager cameraManager;
-
-  @SuppressLint("UseSparseArrays")
-  private static Map<Long, Cam> cams = new HashMap<>();
-
-  static {
-    ORIENTATIONS.append(Surface.ROTATION_0, 0);
-    ORIENTATIONS.append(Surface.ROTATION_90, 90);
-    ORIENTATIONS.append(Surface.ROTATION_180, 180);
-    ORIENTATIONS.append(Surface.ROTATION_270, 270);
-  }
+  private static final int CAMERA_REQUEST_ID = 513469796;
+  private static final String TAG = "CameraPlugin";
+  private static final SparseIntArray ORIENTATIONS =
+      new SparseIntArray() {
+        {
+          append(Surface.ROTATION_0, 0);
+          append(Surface.ROTATION_90, 90);
+          append(Surface.ROTATION_180, 180);
+          append(Surface.ROTATION_270, 270);
+        }
+      };
 
+  private static CameraManager cameraManager;
   private final FlutterView view;
+  private Camera camera;
```

Comment:
It was making things a lot more complicated, the current code would need a lot of changes to handle multiple cameras. Besides, you can't have more than one camera open at one time on Android. I don't think it's worst the effort until Android can handle more than one camera at a time.

Reply:
@sigurdm What do you mean by that?

Reply:
This changes the plugin to only have one Camera, instead of a map of Camera s.
This could be done similarly on the ios side (there is a map `_cams`).

Reply:
The variable _cams, is a dictionary with capacity 1. As in, at any point it maps one texture ID to one camera. It does not contain multiple cameras at one point.